src/api/app/assets/javascripts/webui2/request.js

```diff
@@ -0,0 +1,211 @@
+ minChars: 0, matchCase: true, max: 50, extraParams: {project: function() { return $("#target_project").val(); }}
+ });*/
+
+function requestAddAcceptRequestButton() { // jshint ignore:line
+  $('#accept_request_button').click(function () {
+    var additionalElement;
+
+    /* Add some hidden elements to carry HTML form data that is found at other DOM places for design reasons.  */
+    if ($('.submitter_is_maintainer_checkbox').size() !== 0 &&
+      $('.submitter_is_maintainer_checkbox').is(':checked')) {
+      additionalElement = '<input id="' + $('.submitter_is_maintainer_checkbox').attr('id') +
+        '" name="' + $('.submitter_is_maintainer_checkbox').attr('name') +
+        '" type="hidden" value="' + $('.submitter_is_maintainer_checkbox').attr('value') + '"/>';
+      $('#request_handle_form p:last').append(additionalElement);
+    }
+    if ($('.forward_checkbox').size() !== 0 &&
+      $('.forward_checkbox').is(':checked')) {
+      $('.forward_checkbox').each(function () {
+        additionalElement = '<input id="' + $(this).attr('id') +
+          '" name="' + $(this).attr('name') +
+          '" type="hidden" value="' + $(this).attr('value') + '"/>';
+        $('#request_handle_form p:last').append(additionalElement);
+      });
+    }
+  });
+}
+
+function requestShowReview() { // jshint ignore:line
+  var index;
+  $('.review_descision_link').click(function (event) {
+    $('#review_descision_select li.selected').attr('class', '');
+    $(event.target).parent().attr('class', 'selected');
+    $('.review_descision_display').hide();
+    index = event.target.id.split('review_descision_link_')[1];
+    $('#review_descision_display_' + index).show();
+    return false;
+  });
+}
+
+function requestAddReviewAutocomplete() { // jshint ignore:line
+
+  $('#review_type').change(function () {
+    switch ($('#review_type option:selected').attr('value')) {
+      case "user":
+      {
+        $('#review_user_span').show();
+        $('#review_group_span').hide();
+        $('#review_project_span').hide();
+        $('#review_package_span').hide();
+      }
+        break;
+      case "group":
+      {
+        $('#review_user_span').hide();
+        $('#review_group_span').show();
+        $('#review_project_span').hide();
+        $('#review_package_span').hide();
+      }
+        break;
+      case "project":
+      {
+        $('#review_user_span').hide();
+        $('#review_group_span').hide();
+        $('#review_project_span').show();
+        $('#review_package_span').hide();
+      }
+        break;
+      case "package":
+      {
+        $('#review_user_span').hide();
+        $('#review_group_span').hide();
+        $('#review_project_span').show();
+        $('#review_package_span').show();
+      }
+        break;
+    }
+  });
+
+  $("#review_group").autocomplete({source: '/group/autocomplete', minChars: 2, matchCase: true, max: 50,
```

Comment:
This could be abstracted in a function, it's always the same code (beside a few different strings)